### PR TITLE
tools: Add ICS html-to-rst file parser

### PR DESCRIPTION
**Pull request by mkasenberg**
The tool parses html file, saved form page https://launchstudio.bluetooth.com/ICS/<some-id>, that contains ICS tables, into rst file.

### CODE REVIEW

**Review by sjanc — APPROVED** [review]

File: tools/ics_rst_from_html.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+
+def table_div(max_cols, header_flag=1):
+    out = ""
+    if header_flag == 1:
+        style = "="
+    else:
+        style = "-"
+
+    for max_col in max_cols:
```
Comment (michal-narajowski) [inline tools/ics_rst_from_html.py:70]:
```suggestion
    for col_width in max_width_per_column:
```

File: tools/ics_rst_from_html.py [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    out = out.strip() + "\n"
+    return out
+
+
+def normalize_row(row, max_cols):
+    r = ""
+    for i, max_col in enumerate(max_cols):
+        try:
+            r += row[i] + (max_col - len(row[i]) + 1) * " "
+        except Exception as e:
```
Comment (michal-narajowski) [inline tools/ics_rst_from_html.py:82]: Do we need this here? If there is a possibility of an exception we should implement this in such a way as to not trigger it. And definitely not handle generic Exception, at least a specific exception that is triggered.

File: tools/ics_rst_from_html.py [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+    return r.strip() + "\n"
+
+
+if __name__ == '__main__':
+
+    if len(sys.argv) < 3:
+        sys.exit('Please pass profile name and html file path as arguments, e.g.:\n'
+                 '$ python3 %s GAP path/to/Launch-Studio-ICS-Selection.html\n'
+                 'html file - save https://launchstudio.bluetooth.com/ICS/<some id> '
+                 'page as html file' % sys.argv[0])
```
Comment (michal-narajowski) [inline tools/ics_rst_from_html.py:94]: Use new formatting syntax.

File: tools/ics_rst_from_html.py [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+
+    table_titles = re.findall(r'(?=<h4>\s*Table\s*.+:\s*).+?(?=</h4>)', content)
+    for i in range(len(table_titles)):
+        table_titles[i] = table_titles[i].split(r':')[1].strip()
+
+    with open(temp_path, 'w') as f:
+        f.write(content)
+
+    # Returns list of all tables on page
+    tables = pd.read_html(temp_path)
```
Comment (michal-narajowski) [inline tools/ics_rst_from_html.py:128]: `pd.read_html()` also takes raw HTML string so no need to create temp file.

File: tools/ics_rst_from_html.py [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+TCRL version: %s
+
+M - mandatory
+
+O - optional
+
+
+''' % (profile_name.lower(), head, len(head) * '*', tcrl_ver)
+
+    for tab, title in zip(tables, table_titles):
```
Comment (michal-narajowski) [inline tools/ics_rst_from_html.py:147]:
```suggestion
    for table, table_title in zip(tables, table_titles):
```

File: tools/ics_rst_from_html.py [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+In directory of run will be created gap-pics.rst file.
+"""
+
+import sys
+import os
+import re
+import pandas as pd
+
+
+def make_table(_tab, _profile):
```
Comment (michal-narajowski) [inline tools/ics_rst_from_html.py:37]:
```suggestion
def make_table(table, profile):
```

**Review by michal-narajowski — CHANGES_REQUESTED** [review]
Nice work on the script! I have some comments but otherwise looks good.

File: tools/ics_rst_from_html.py (hunk 2 above)
Comment (mkasenberg) [reply tools/ics_rst_from_html.py]: Sorry, it's just debugging leftovers.

**Review by michal-narajowski — APPROVED** [review]

**Pull request merged**